crates/proto/src/udp/udp_stream.rs

```diff
@@ -93,6 +93,16 @@ impl<P: RuntimeProvider> UdpStream<P> {
     /// * `remote_addr` - socket address for the remote connection (used to determine IPv4 or IPv6)
     /// * `bind_addr` - optional local socket address to connect from (if a nonzero port number is
     ///                 specified, it will be used instead of randomly selecting a port)
+    /// * `os_port_selection` - optional parameter to specify whether to use the operating system's
```

Comment:
Nits: I don't think we use double-space-after-period style elsewhere?

For "mutually exclusive", I think it would be better to explicit explain how they interact, maybe like "If os_port_selection is active, `avoid_local_parts` is ignored."